src/compute-client/src/controller/instance.rs

```diff
@@ -1232,20 +1245,26 @@ where
             // to run again.
             true
         } else {
-            // Check dependency frontiers to determine if all inputs are available.
-            // An input is available when its frontier is greater than the `as_of`, i.e., all input
-            // data up to and including the `as_of` has been sealed.
+            // Check dependency frontiers to determine if all inputs are
+            // available. An input is available when its frontier is greater
+            // than the `as_of`, i.e., all input data up to and including the
+            // `as_of` has been sealed.
             let compute_frontiers = collection.compute_dependencies.iter().map(|id| {
                 let dep = &self.compute.expect_collection(*id);
                 &dep.write_frontier
             });
-            let storage_frontiers = collection.storage_dependencies.iter().map(|id| {
-                let dep = &self.storage_controller.collection(*id).expect("must exist");
-                &dep.write_frontier
-            });
-            compute_frontiers
+
+            let storage_frontiers = self
+                .storage_controller
+                .collections_frontiers(collection.storage_dependencies.iter().cloned().collect())
```

Comment:
```suggestion
                .collections_frontiers(collection.storage_dependencies.clone()))
```

I think that should work?

Reply:
facepalm!